src/components/connectAdvanced.js

```diff
@@ -86,6 +91,11 @@ export default function connectAdvanced(
     [subscriptionKey]: subscriptionShape,
   }
 
+  function getDerivedStateFromProps(nextProps, prevState) {
+    prevState.selector.run(nextProps)
```

Comment:
This looks mutative. What does it actually do?

I'm worried that this won't be safe and then avoiding the warning just gives a false sense of security.

Reply:
It's not pure but it's idempotent.
I can try tomake it pure by moving all the fields from `selector` to  state though.

Reply:
Let's try that!

Reply:
Did that. It required some minor changes in tests: https://github.com/reactjs/react-redux/pull/919/commits/a7ba2981b4e4482830da85b105d86600811c7938#diff-b5fd7a42ea1b8efa382416b4a323003c

It's quite clear why `setState` calls count changed, but as for `mapStateToPropsCalls`, I'm not so confident